packages/core/src/utils/serializeNode.tsx

```diff
@@ -31,7 +31,7 @@ export const serializeComp = (
         }
         return serializeComp(child, resolver);
       });
-    } else if (prop.type) {
+    } else if (typeof prop.type === 'function') {
```

Comment:
Hm, this only works if the component is wrapped inside an object literal:
```tsx
<SomeComponent someObj={{ type: SomeOtherComponent }} />
```

But what about:
```tsx
<SomeComponent a={SomeOtherComponent} />
```

I actually can't really think of a strong use case where we need to serialise a component that's being passed as a prop. Perhaps, we should just ignore serialising any props that are not serialisable, including any components? So if we hit a `prop` that's a typeof function, we will just ignore it and move on to the next prop.

Reply:
Yeah, skipping serializing any props that are functions makes sense to me. For this case though since the `type` prop is added internally by craft.js it seems like this PR should be enough to fix the issue, right?

Or what would the fix for handling any functions be?
I imagine, maybe just changing to something like this:

```
if (prop === undefined || prop === null || typeof prop === 'function') {
      return result;
}
```

Reply:
Yes, your solutions makes sense. 

I think we can skip serialising the `children` prop as well (since we're treating it the same way as functions right now). I feel like allowing this serialisation has always been troublesome.

https://github.com/prevwong/craft.js/blob/47336afd2c262312c093877409de249cb3683c67/packages/core/src/utils/serializeNode.tsx#L27-L35

Essentially, any prop that's not a JS literal (string, number, array, object etc) should be ignored. Ie:

```tsx
<SomeComponent 
  color="#333" // ok
  count={2} // ok
  values={{ someKey: 2, }} // ok
  someComp={AnotherComponent} // ignored
  someArray={[1,2,3]} // ok
  someComplexArray={[AnotherComponent]} // ignored
>
  Some Text // ok
  <AnotherComponent /> // ignored
</SomeComponent>
```

But perhaps this might need a bit more work to do. 

For now, we can just do the quick fix and ignore if the prop is a function.